src/background/stores/assessment-card-selection-store.ts

```diff
@@ -0,0 +1,231 @@
+            !this.state[payload.testKey].rules?.[payload.ruleId] ||
+            this.state[payload.testKey].rules![payload.ruleId].cards[payload.resultInstanceUid] ===
+                undefined
+        ) {
+            return;
+        }
+
+        const rule = this.state[payload.testKey].rules![payload.ruleId];
+        const isSelected = !rule.cards[payload.resultInstanceUid];
+        rule.cards[payload.resultInstanceUid] = isSelected;
+
+        // whenever a card is selected, the visual helper is enabled
+        if (isSelected) {
+            this.state[payload.testKey].visualHelperEnabled = true;
+            this.state[payload.testKey].focusedResultUid = payload.resultInstanceUid;
+        }
+
+        await this.emitChanged();
+    };
+
+    private collapseAllRules = async (payload: AssessmentExpandCollapsePayload): Promise<void> => {
+        if (
+            !payload ||
+            !payload.testKey ||
+            !this.state[payload.testKey] ||
+            !this.state[payload.testKey].rules
+        ) {
+            return;
+        }
+
+        forOwn(this.state[payload.testKey].rules!, rule => {
+            rule.isExpanded = false;
+            this.deselectAllCardsInRule(rule);
+        });
+
+        await this.emitChanged();
+    };
+
+    private expandAllRules = async (payload: AssessmentExpandCollapsePayload): Promise<void> => {
+        if (
+            !payload ||
+            !payload.testKey ||
+            !this.state[payload.testKey] ||
+            !this.state[payload.testKey].rules
+        ) {
+            return;
+        }
+
+        forOwn(this.state[payload.testKey].rules!, rule => {
+            rule.isExpanded = true;
+        });
+
+        await this.emitChanged();
+    };
+
+    private toggleVisualHelper = async (
+        payload: AssessmentCardToggleVisualHelperPayload,
+    ): Promise<void> => {
+        if (!payload || !payload.testKey || !this.state[payload.testKey]) {
+            return;
+        }
+
+        this.state[payload.testKey].visualHelperEnabled =
+            !this.state[payload.testKey].visualHelperEnabled;
+
+        if (!this.state[payload.testKey].visualHelperEnabled) {
+            this.deselectAllCards();
+        }
+
+        await this.emitChanged();
+    };
+
+    private onResetFocusedIdentifier = async (
+        payload: AssessmentResetFocusedIdentifierPayload,
+    ): Promise<void> => {
+        if (!payload || !payload.testKey || !this.state[payload.testKey]) {
+            return;
+        }
+
+        this.state[payload.testKey].focusedResultUid = null;
+        await this.emitChanged();
+    };
+
+    private onNavigateToNewCardsView = async (
+        payload: AssessmentNavigateToNewCardsViewPayload,
+    ): Promise<void> => {
+        if (!payload || !payload.testKey || !this.state[payload.testKey]) {
+            return;
+        }
+
+        this.state[payload.testKey].focusedResultUid = null;
+
+        if (this.state[payload.testKey].rules) {
```

Comment:
Random musing: since a lot of this logic seems to be duplicated across all the card selection stores, it might make sense to refactor to consolidate to one place, maybe make the card selection store an abstract class with this logic instead of an interface. 
For the purposes of this PR/ feature, I don't want to block on it, but might be something worthwhile to file an engineering issue over.